level1c4pps/__init__.py

```diff
@@ -55,6 +88,146 @@ def dt64_to_datetime(dt64):
+        # pygac qual flags
+        enc = {'dtype': 'int16', 'zlib': True,
+               'complevel': 4, '_FillValue': -32001.0}
+    if name in ['scanline_timestamps']:
+        # pygac scanline_timestamps
+        enc = {'dtype': 'int64', 'zlib': True,
+               'complevel': 4, '_FillValue': -1.0}
+    return name, enc
+
+
+def rename_latitude_longitude(scene):
+    """Rename latitude longitude to lat lon."""
+    scene['lat'] = scene['latitude']
+    scene['lon'] = scene['longitude']
+    del scene['latitude']
+    del scene['longitude']
+    # Update attributes
+    scene['lat'].attrs['long_name'] = 'latitude coordinate'
+    scene['lon'].attrs['long_name'] = 'longitude coordinate'
+    try:
+        del scene['lat'].coords['acq_time']
+        del scene['lon'].coords['acq_time']
+    except KeyError:
+        pass
+
+
+def update_angle_attributes(scene, start_time):
+    """Set and delete angle attributes."""
+    for angle in ['sunzenith', 'satzenith', 'azimuthdiff', 'sunazimuth', 'satazimuth']:
+        if angle not in scene.keys() and angle in ['sunazimuth', 'satazimuth']:
+            # azimuth angles not always there
+            continue
+        scene[angle].attrs['id_tag'] = angle
+        scene[angle].attrs['name'] = angle
+        scene[angle].attrs['coordinates'] = 'lon lat'
+        scene[angle].attrs['long_name'] = ANGLE_ATTRIBUTES['long_name'][angle]
+        scene[angle].attrs['valid_range'] = ANGLE_ATTRIBUTES['valid_range'][angle]
+        scene[angle].attrs['standard_name'] = ANGLE_ATTRIBUTES['standard_name'][angle]
+        scene[angle].coords['time'] = start_time
+        # delete some attributes
+        del scene[angle].attrs['area']
+        try:
+            del scene[angle].coords['acq_time']
+        except KeyError:
+            pass
+
+
+def compose_filename(scene, out_path, instrument, band=None):
+    """Compose output filename.
+
+    As default use the start and end time of the scene.
+    For SEVIRI this is the nominal timestamp of the scan (as in the HRIT files).
+    If a scene band is supplied use that for start/end time.
+
+    Args:
+        scene: satpy scene
+        outpath: output directory (string)
+        instrument: lower case instrument (string)
+        band: use start and end time from band if supplied
+
+    """
+    start_time = scene.attrs['start_time']
+    end_time = scene.attrs['end_time']
+    if band is not None:
+        start_time = band.attrs['start_time']
+        end_time = band.attrs['end_time']
+    platform_name = scene.attrs['platform']
+    orbit_number = int(scene.attrs['orbit_number'])
+    filename = os.path.join(
+        out_path,
+        "S_NWC_{:s}_{:s}_{:05d}_{:s}Z_{:s}Z.nc".format(
+            instrument,
+            platform_name.lower().replace('-', ''),
+            orbit_number,
+            datetime.strftime(dt64_to_datetime(start_time), '%Y%m%dT%H%M%S%f')[:-5],
+            datetime.strftime(dt64_to_datetime(end_time), '%Y%m%dT%H%M%S%f')[:-5]))
```

Comment:
Are you aware of the excellent `trollsift` library which is made just for that ? 
https://trollsift.readthedocs.io/en/latest/usage.html#parser

Reply:
Now I am, thanks!